_drafts/circle-ci-chibios.md

```diff
@@ -0,0 +1,438 @@
+
+The subset of the syntax we’ll need here is pretty simple:
+```yaml
+# not a valid CircleCI file
+version: 2
+jobs:
+	<job>:
+		<environment>
+		steps:
+			- run:
+				name: <name>
+				command: <shell command>
+			- run:
+				name: <name>
+				command: <shell command>
+	<job>:
+		...
+```
+
+We can fill our steps by simply copying the commands from our bash script
+
+```yaml
+# not a valid CircleCI file
+version: 2
+jobs:
+  build:
+    <environment>
+    steps:
+      - run:
+			name: clone
+			command: git clone git@github.com:memfault/ChibiOS.git
+
+		- run:
+			name: checkout
+			command: git checkout <revision>
+
+      - run:
+          name: Greeting
+          command: echo Hello, world.
+
+      - run:
+          name: Print the Current Time
+          command: date
+```
+
+You will note that we need to know what revision to build in order to run this script. Rather than use a variable, CircleCI has a built-in step called `checkout`[^3] you can use which automatically does the right thing:
+
+```yaml
+# Not a valid CircleCI file
+version: 2
+jobs:
+  build:
+    <environment>
+
+    steps:
+      - checkout
+
+      - run:
+          name: Greeting
+          command: echo Hello, world.
+
+      - run:
+          name: Print the Current Time
+          command: date
+```
+
+All that is left is setting the environment. This section defines what system image is used to run your job. The simplest thing to do here is to set a docker image from the docker registry as your base image. Any docker image will do, in this case we will use the latest Debian image: `debian:stretch`.
+
+Our hello world example is now complete:
+
+```yaml
+version: 2
+jobs:
+  build:
+    docker:
+      - image: debian:stretch
+
+    steps:
+      - checkout
+
+      - run:
+          name: Greeting
+          command: echo Hello, world.
+
+      - run:
+          name: Print the Current Time
+          command: date
+```
+
+Save this file in your repository as `.circleci/config.yml`, and push it:
+
+```
+$ git commit -m 'Add sample .yml CircleCI file'
+[master c347b244e] Add sample .yml CircleCI file
+ 1 file changed, 16 insertions(+)
+ create mode 100644 .circleci/config.yml
+
+$ git push memfault master
+Enumerating objects: 5, done.
+Counting objects: 100% (5/5), done.
+Delta compression using up to 8 threads
+Compressing objects: 100% (3/3), done.
+Writing objects: 100% (4/4), 473 bytes | 473.00 KiB/s, done.
+Total 4 (delta 1), reused 0 (delta 0)
+remote: Resolving deltas: 100% (1/1), completed with 1 local object.
+To github.com:memfault/ChibiOS.git
+   fd444de36..c347b244e  master -> master
+```
+
+Now head to circleci.com and observe that a new job has started:
+
+![](img/circle-ci/sample_build_started.png)
+
+Wait a bit, and you can look at the build result:
+
+![](img/circle-ci/sample_build_ended.png)
+
+You should see the output of your two commands, and a successful result.
+
+### Testing CircleCI Configs Locally
+
```

Comment:
Since CircleCI is just using Docker as the build environment, we can test our configuration locally using the `circleci` CLI tool.